legate/driver/args.py

```diff
@@ -42,6 +46,45 @@
 __all__ = ("parser",)
 
 
+def detect_multi_node_defaults() -> tuple[dict[str, Any], dict[str, Any]]:
+    from os import getenv
+
+    nodes_kw = dict(NODES.kwargs)
+    ranks_per_node_kw = dict(RANKS_PER_NODE.kwargs)
+    where = None
+
+    if nodes_env := getenv("OMPI_COMM_WORLD_SIZE"):
```

Comment:
I realized as I was reading your testcase that I actually made a mistake in my original ask; `OMPI_COMM_WORLD_SIZE` and `MV2_COMM_WORLD_SIZE` don't refer to number of nodes, but total number of ranks; therefore this blob and the next should do something like:

```
    if ranks_env := getenv("OMPI_COMM_WORLD_SIZE"):
         if ranks_per_node_env := getenv("OMPI_COMM_WORLD_LOCAL_SIZE"):
            ranks, ranks_per_node = int(ranks_env), int(ranks_per_node_env)
            if ranks % ranks_per_node != 0:
                raise ValueError(
                    "Detected incompatible ranks and ranks-per-node from the environment"
                )
            nodes = ranks // ranks_per_node
            where = "OMPI"
```

Reply:
https://github.com/nv-legate/legate.core/pull/832/commits/2975cc7ab7304e8f068a2707096ced9c12e4486e